app/routers/quizzes.py

```diff
@@ -3,23 +3,53 @@
 from fastapi.encoders import jsonable_encoder
 from database import client
 from models import Quiz, QuizResponse
+from settings import Settings
 
 router = APIRouter(prefix="/quiz", tags=["Quiz"])
+settings = Settings()
 
 
 @router.post("/", response_model=QuizResponse)
 async def create_quiz(quiz: Quiz):
     quiz = jsonable_encoder(quiz)
-    new_quiz = client.quiz.quizzes.insert_one(quiz)
-    created_quiz = client.quiz.quizzes.find_one({"_id": new_quiz.inserted_id})
 
-    for question_set in created_quiz["question_sets"]:
+    for question_set_index, question_set in enumerate(quiz["question_sets"]):
         questions = question_set["questions"]
-        for index, _ in enumerate(questions):
-            questions[index]["question_set_id"] = question_set["_id"]
+        for question_index, _ in enumerate(questions):
+            questions[question_index]["question_set_id"] = question_set["_id"]
 
         client.quiz.questions.insert_many(questions)
 
+        subset_with_all_details = client.quiz.questions.aggregate(
```

Comment:
based on the name of the variable below, this seemed more appropriate, no?
```suggestion
        subset_with_details = client.quiz.questions.aggregate(
```